JIT: Simplify JitDisasm matching behavior and support namespaces/generics in release (#74430)
dotnet/runtime: docs/design/coreclr/jit/viewing-jit-dumps.md

## Specifying method names

> Some environment variables such as `DOTNET_JitDump` take a set of patterns specifying method names. The matching works in the following way:

Reviewer: At some point (but not in this change) we should probably make a complete change of our documentation to use the DOTNET_ form instead of the COMPlus_ form, but it's odd to only change this one case in this file.

Author: Sure, I'll revert this.

```suggestion
Some environment variables such as `COMPlus_JitDump` take a set of patterns specifying method names. The matching works in the following way:
```

> Some environment variables such as `COMPlus_JitDump` take a set of patterns specifying method names. The matching works in the following way:

Reviewer 2: ```suggestion
Some environment variables such as `DOTNET_JitDump` take a set of patterns specifying method names. The matching works in the following way:
```

* The environment variable is a space-separated list of patterns that can be quoted if they need to contain spaces.
* Patterns can contain * and ? wildcards matching respectively any characters and any 1 character.

> * If the pattern contains a : character, then the string matched against is prefixed with `ClassName:`.

Reviewer 2: ```suggestion
* If the pattern contains a ':' character, then the string matched against is prefixed with `ClassName:`.
```

> * If the patter ncontains a ( character, then the string matched against is suffixed with its signature.

Reviewer 2: ```suggestion
* If the pattern contains a '(' character, then the string matched against is suffixed with its signature.
```

## Specifying method names

Some environment variables such as `COMPlus_JitDump` take a set of patterns specifying method names. The matching works in the following way:

> * The JitDisasm string is a space-separated list of patterns. Patterns can arbitrarily contain both '*' (match any characters) and '?' (match any 1 character).

Reviewer: nit
```suggestion
* A method set string is a space-separated list of patterns. Patterns can arbitrarily contain both '*' (match any characters) and '?' (match any 1 character).
```
